src/ast/macros.rs

```diff
@@ -0,0 +1,327 @@
+/// Implement the node trait.
+macro_rules! Node {
```

Comment:
I haven't yet finished reviewing,
so I don't know what the final module layout should look like,
but I'm not sure if the benefit to moving macros to a separate
file is worth tearing it apart from ast.rs.
IIRC, none can live without the other, so I'm not sure if it helps.

Of course we do want to look at extracting a fish-syntax crate in
future, which should be able to parse fish code.
So we should do whatever helps but this seems unrelated